raft/src/main/java/org/apache/kafka/raft/RaftUtil.java

```diff
@@ -39,4 +47,39 @@ public static ByteBuffer serializeRecords(Records records) throws IOException {
         }
     }
 
+    public static ApiMessage errorResponse(ApiKeys apiKey, Errors error) {
+        switch (apiKey) {
+            case VOTE:
+                return new VoteResponseData()
```

Comment:
A side note, I checked the current implementation of all RPCs and found that most response RPC contains the top level error code. For response using partition level error code, the hierarchy could be revised by separating top level error code and partition level error. Maybe we could make it as a sub-task for https://issues.apache.org/jira/browse/KAFKA-9993 to create a `ApiResponse` subclass that always contains an error code, which I believe would simplify boiler templates a lot. cc @guozhangwang

Reply:
Yes, this would allow us to cleanup some of this logic. Probably not a huge priority, but worth doing eventually. I think more stuff like this will emerge when we are ready to try and replace the existing Request/Response objects with the generated types.